### PR TITLE
convert to hpack, add cabal synopsis and stuff, add nix files

### PR DESCRIPTION
**Pull request by phunehehe**
(cut out from https://github.com/zalora/upcast/pull/44)

### CODE REVIEW

File: package.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+copyright: 2014, 2015 Zalora South East Asia Pte Ltd
+license: MIT
+
+synopsis:    Nix-based Linux deployment platform tools
+description: Nix-based Linux deployment platform tools
+
+data-files:
+  - nix/*.nix
+  - nix/aws/*.nix
+  - nix/lib/*.nix
```
Comment (sol) [inline package.yaml]: Try `**/*.nix`

File: package.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  - time
+  - unix
+  - unordered-containers
+  - vector
+  - vk-posix-pty >= 0.2.1
+  - xml-conduit
+
+library:
+    source-dirs: src
+    default-extensions: OverloadedStrings
```
Comment (sol) [inline package.yaml]: This can be further DRYed up by moving the previous two lines to the top level.

Comment (phunehehe) [inline package.yaml]: thanks, updated

**Comment by sol** [comment]
If you plan to release this yo Hackage you may also want to add `github: zalora/upcast`.

**Comment by phunehehe** [comment]
I think I already have that?

**Comment by sol** [comment]
Ah, ic, somehow I missed that...

**Pull request merged**